### PR TITLE
Add flutter 1.20.2

### PR DESCRIPTION
**Pull request by andreiborisov**
After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

**Comment by andreiborisov** [comment]
This PR is updated version of https://github.com/Homebrew/homebrew-cask/pull/87344 to speed things up while the original author is not available

File: Casks/flutter.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  # storage.googleapis.com/flutter_infra/releases/stable/macos/ was verified as official when first introduced to the cask
+  url "https://storage.googleapis.com/flutter_infra/releases/stable/macos/flutter_macos_#{version}-stable.zip"
+  name "Flutter SDK"
+  desc "UI toolkit for building applications for mobile, web and desktop"
+  homepage "https://flutter.dev/"
+
+  # Flutter manages its components by itself so it should not be updated by
+  # Homebrew in most cases, else those components would be lost on upgrade
+  auto_updates true
+  conflicts_with cask: "dart"
```
Comment (reitermarkus) [inline Casks/flutter.rb:14]: There is no `dart` cask.

Comment (andreiborisov) [reply Casks/flutter.rb:14]: There is an official one from Dart project tap though: https://github.com/dart-lang/homebrew-dart

Should I delete the field anyway?

File: Casks/flutter.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  depends_on macos: ">= :catalina"
+
+  binary "flutter/bin/dart"
+  binary "flutter/bin/flutter"
+
+  # Not actually necessary, since it would be deleted anyway. It is present to
+  # make clear an uninstall was not forgotten and that for this cask it is
+  # indeed this simple.
+  uninstall delete: "#{staged_path}/#{token}"
```
Comment (reitermarkus) [inline Casks/flutter.rb:23]: Remove this. We only require this to be explicitly stated for `installer` casks.

Comment (andreiborisov) [reply Casks/flutter.rb:22]: 👌

**Pull request merged**